### PR TITLE
misc changes

### PR DESCRIPTION
**Pull request by srajangarg**

### CODE REVIEW

File: symengine/polys/upolybase.h [hunk 1]
```diff
@@ -506,6 +507,26 @@ RCP<const Poly> quo_upoly(const Poly &a, const Poly &b)
...
+    typename Poly::container_type res(1);
+
+    while (p != 1) {
+        if (p % 2 == 0) {
+            tmp = tmp * tmp;
+            p >>= 1;
+        } else {
+            res = res * tmp;
+            tmp = tmp * tmp;
+            p = (p - 1) / 2;
```
Comment (nishnik) [inline symengine/polys/upolybase.h]: This seems better: `p >>= 1`, outside of the `if-else` block

**Comment by srajangarg** [comment]
@isuruf ping

**Comment by certik** [comment]
@srajangarg can you please merge with master and resolve conflicts?

File: symengine/pow.cpp [hunk 2]
```diff
@@ -147,6 +147,9 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
             ->power_num(outArg(coef), d, rcp_static_cast<const Number>(b));
         return Mul::from_dict(coef, std::move(d));
     }
+    if (is_a<const Log>(*b) and eq(*a, *E))
+        return rcp_static_cast<const Log>(b)->get_arg();
```
Comment (isuruf) [inline symengine/pow.cpp]: To enforce this, there are some other places where you have to check this. For example, `E**(log(x) + 1) / E` should not be `E**log(x)`, but `x`

**Comment by certik** [comment]
Did github remove an option to comment on a whole commit?

**Comment by certik** [comment]
Regarding 695f45e133b268f0a9b2218c2413766213a4f34b, @srajangarg can you explain in the git commit log:

* what's the motivation for the refactoring
* are you adding some new logic into the code? Fixing some bugs?
* speed implications?

The commit ff699d07c91f001c1715dd24fb40caa8cc615c17 needs a test.

Overall it looks like this PR is a bunch of unrelated changes, so they should be all sent in separate PRs.

**Comment by srajangarg** [comment]
Okay, I think I'll open other PR(s).

**Comment by Sumith1896** [comment]
Some changes are subsumed here https://github.com/symengine/symengine/pull/1004 
New PR would be great, thanks!

**Pull request closed without merging**